src/components/tile/tile-content/tile-content.test.tsx

```diff
@@ -1,143 +1,51 @@
-      });
-
-      expect(screen.getByText("Tile Content 3")).toHaveStyle({
-        "padding-left": leftValue,
-        "padding-right": "0px",
-      });
-    }
-  );
-
-  it.each([
-    [{}, "var(--spacing300)", undefined],
-    [{ p: 2 }, "var(--spacing200)", undefined],
-    [{ pt: 2, p: 1 }, "var(--spacing200)", "var(--spacing100)"],
-    [{ pb: 2, p: 1 }, "var(--spacing100)", "var(--spacing200)"],
-    [{ py: 2, p: 1 }, "var(--spacing200)", undefined],
-  ] as [PaddingProps, string, string | undefined][])(
-    "applies the expected padding values when %s passed and in vertical orientation",
-    (paddingPropsFromTile, topValue, bottomValue) => {
-      render(
-        <>
-          <TileContext.Provider
-            value={{ isHorizontal: false, paddingPropsFromTile }}
-          >
-            <TileContent>Tile Content 1</TileContent>
-            <TileContent>Tile Content 2</TileContent>
-            <TileContent>Tile Content 3</TileContent>
-          </TileContext.Provider>
-        </>
-      );
+testStyledSystemSpacingRTL(
+  (props) => (
+    <TileContent data-role="tile-content" {...props}>
+      Test
+    </TileContent>
+  ),
+  () => screen.getByTestId("tile-content")
+);
+testStyledSystemWidthRTL(
+  (props) => (
+    <TileContent data-role="tile-content" {...props}>
+      Test
+    </TileContent>
+  ),
+  () => screen.getByTestId("tile-content")
+);
+testStyledSystemHeightRTL(
+  (props) => (
+    <TileContent data-role="tile-content" {...props}>
+      Test
+    </TileContent>
+  ),
+  () => screen.getByTestId("tile-content")
+);
+
+test("does not render when no children are passed", () => {
+  render(<TileContent data-role="tile-content" />);
+
+  expect(screen.queryByTestId("tile-content")).not.toBeInTheDocument();
+});
 
-      expect(screen.getByText("Tile Content 1")).toHaveStyle({
-        "padding-top": "0px",
-        "padding-bottom": bottomValue || topValue,
-      });
+test("does not render when falsy children are passed", () => {
+  render(<TileContent data-role="tile-content">{null}</TileContent>);
 
-      expect(screen.getByText("Tile Content 2")).toHaveStyle({
-        "padding-top": topValue,
-        "padding-bottom": bottomValue || topValue,
-      });
+  expect(screen.queryByTestId("tile-content")).not.toBeInTheDocument();
+});
 
-      expect(screen.getByText("Tile Content 3")).toHaveStyle({
-        "padding-top": topValue,
-        "padding-bottom": "0px",
-      });
-    }
-  );
+test("renders when children are passed", () => {
+  render(<TileContent>Tile Content</TileContent>);
 
-  it("has proper data attributes applied", () => {
-    render(
-      <TileContent data-element="foo" data-role="bar">
-        Tile Content
-      </TileContent>
-    );
-    const element = screen.getByText("Tile Content");
-    expect(element).toHaveAttribute("data-component", "tile-content");
-    expect(element).toHaveAttribute("data-element", "foo");
-    expect(element).toHaveAttribute("data-role", "bar");
-  });
```

Comment:
**suggestion**: would it be worth us keeping this test? If consumers are targeting `data`-* tags in their UI tests, it might be beneficial for us to assert we haven't accidentally broken this.